### PR TITLE
PubKeyFromJWK should only accept JWKs containing public keys

### LINKED ISSUE #18: PubKeyFromJWK assumes a private key
Currently `PubKeyFromJWK` assumes that the passed JWK contains a private key.

In turn, this means that `evcli <XYZ> check [...]` cannot be passed a pure public key (as it should).

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #18

### CODE REVIEW

File: cmd/cca/check_test.go [hunk 1]
```diff
@@ -192,3 +192,26 @@ func Test_CheckCmd_token_invalid_format(t *testing.T) {
...
+	require.NoError(t, err)
+
+	err = afero.WriteFile(fs, "es256.jwk", testValidPAK, 0644)
+	require.NoError(t, err)
+
+	cmd := NewCheckCmd(fs)
+	cmd.SetArgs(
+		[]string{
+			"--token=ccatoken.cbor",
+			"--key=es256.jwk",
```
Comment (yogeshbdeshpande) [inline cmd/cca/check_test.go:209]: --key should be 384 right?

Comment (yogeshbdeshpande) [reply cmd/cca/check_test.go:209]: For CCA?

File: cmd/psa/check_test.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ func Test_CheckCmd_ok(t *testing.T) {
 	err := afero.WriteFile(fs, "psatoken.cbor", testValidP2PSAToken, 0644)
 	require.NoError(t, err)
 
-	err = afero.WriteFile(fs, "es256.jwk", testValidKey, 0644)
+	err = afero.WriteFile(fs, "es256.jwk", testValidKeyPub, 0644)
```
Comment (yogeshbdeshpande) [inline cmd/psa/check_test.go:20]: Ditto!

**Comment by yogeshbdeshpande** [comment]
Thanks for the code change!

I will review this properly later today or tomorrow.

File: cmd/cca/check_test.go (hunk 1 above)
Comment (thomas-fossati) [reply cmd/cca/check_test.go:209]: it really doesn't matter here, it could be called any name

File: cmd/psa/check_test.go (hunk 2 above)
Comment (thomas-fossati) [reply cmd/psa/check_test.go:20]: same, the name here is irrelevant

File: cmd/cca/check_test.go (hunk 1 above)
Comment (thomas-fossati) [inline cmd/cca/check_test.go:209]:
```suggestion
			"--key=not-a-public-key.jwk",
```

File: cmd/psa/check_test.go (hunk 2 above)
Comment (thomas-fossati) [reply cmd/psa/check_test.go:20]: also in this case it's incidentally an ES256 key

File: cmd/cca/check_test.go (hunk 1 above)
Comment (thomas-fossati) [reply cmd/cca/check_test.go]: I've given it a more eloquent name

File: cmd/cca/check_test.go [hunk 3]
```diff
@@ -192,3 +192,26 @@ func Test_CheckCmd_token_invalid_format(t *testing.T) {
...
+
+	cmd := NewCheckCmd(fs)
+	cmd.SetArgs(
+		[]string{
+			"--token=ccatoken.cbor",
+			"--key=not-a-public-key.jwk",
+		},
+	)
+
+	expectedErr := `verifying CCA evidence from ccatoken.cbor using key from not-a-public-key.jwk: unable to verify platform token: unable to instantiate verifier: ES256: algorithm mismatch`
```
Comment (yogeshbdeshpande) [inline cmd/cca/check_test.go:213]: this error does not seem to indicate the type of test which is intended ?

The error says algorithm mismatch however as I understand is that it is not a public key but a provate key used to verify the signature on the token?

**Review by yogeshbdeshpande — COMMENTED** [review]
In general LGTM, left one comment to investigate if we have done any functional error in understanding the interface?

File: cmd/cca/check_test.go (hunk 3 above)
Comment (thomas-fossati) [reply cmd/cca/check_test.go:213]: Yeah, I had the same reaction.  The last bit of the error string is completely misleading.  If it ended with "unable to instantiate verifier" it'd be awesome.  Unfortunately, go-cose wants to be helpful (see https://github.com/veraison/go-cose/blob/07c7d28e2c7eb9d6daa4619b98eceeaf7a8e5156/verifier.go#L47) but misinterprets the situation.

**Comment by thomas-fossati** [comment]
> In general LGTM, left one comment to investigate if we have done any functional error in understanding the interface?

no, we've hit an "error reporting" mishap in go-cose.

**Comment by yogeshbdeshpande** [comment]
> > In general LGTM, left one comment to investigate if we have done any functional error in understanding the interface?
> 
> no, we've hit an "error reporting" mishap in go-cose.

Thanks: I think, looking at this part of the code the returned error (from go-cose) itself is wrong.

We should raise a github issue (go-cose) and fix it.

it should return something like, InvalidPublicKey.

Do you agree?

**Comment by thomas-fossati** [comment]
> > > In general LGTM, left one comment to investigate if we have done any functional error in understanding the interface?
> > 
> > 
> > no, we've hit an "error reporting" mishap in go-cose.
> 
> Thanks: I think, looking at this part of the code the returned error (from go-cose) itself is wrong.
> 
> We should raise a github issue (go-cose) and fix it.
> 
> it should return something like, InvalidPublicKey.
> 
> Do you agree?

sure

File: cmd/cca/check_test.go (hunk 3 above)
Comment (yogeshbdeshpande) [reply cmd/cca/check_test.go:213]: Issue https://github.com/veraison/go-cose/issues/124 raised on go-cose to track this!

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Pull request merged**